### ISSUE TITLE
Core dump when cast Rect

### ISSUE THREAD

**Issue by darkin1** [SerafimArts/ffi-sdl#1]
Whenever when I create `SDL_React`  and then I try to `var_dump` it or set variable for it I got
`50030 segmentation fault (core dumped)  php index.php`

```
$rect = $sdl->new(SDL_Rect::class);
$rect->x = 1;
$rect->y = 1;
$rect->w = 10;
$rect->h = 10;
$sdlRect = $sdl->cast(\Serafim\SDL\RectPtr::class, $rect); // <<<<< HERE
```

```
var_dump($sdlRect);
```
or 
```
$sdlRect->x = 32;
```
or
```
$sdl->SDL_QueryTexture($texture, null, null,  SDL::addr($sdlRect->w), SDL::addr($sdlRect->h) );
```

```
➜ php index.php
object(FFI\CData:struct SDL_Rect*)#21 (1) {
  [0]=>
  [1]    51564 segmentation fault (core dumped)  php index.php
```

**Comment by SerafimArts** [comment]
This is logical. You are trying to cast a structure to its pointer. And during the dump, you try to crawl out of the local memory area.

Other than uncertain behavior and was not expected. Even in this way, you can easily get a "blue screen of death" =))) When you work directly with memory you should be careful

TL;DR; Read the documentation 
1) https://www.php.net/manual/en/ffi.addr.php
2) https://www.php.net/manual/en/ffi.cast.php

**Comment by SerafimArts** [comment]
P.S. 
- `\Serafim\SDL\Rect::class` is an alias of `SDL_Rect`
> `$sdl->new(Rect::class) === $sdl->new('SDL_Rect')`
- `\Serafim\SDL\RectPtr::class` is an alias of `SDL_Rect*`
> `$sdl->cast(RectPtr::class, ...) === $sdl->cast('SDL_Rect *', ...)`
